disperser/apiserver/config.go

```diff
@@ -188,6 +181,88 @@ func ReadCLIConfig(c *cli.Context) (RateConfig, error) {
+		allowlistFile, err := os.Open(allowlistFileName)
+		if err != nil {
+			log.Printf("failed to read allowlist file: %s", err)
+			return allowlist
+		}
+		defer allowlistFile.Close()
+		var allowlistEntries []AllowlistEntry
+		content, err := io.ReadAll(allowlistFile)
+		if err != nil {
+			log.Printf("failed to load allowlist file content: %s", err)
+			return allowlist
+		}
+		err = json.Unmarshal(content, &allowlistEntries)
+		if err != nil {
+			log.Printf("failed to parse allowlist file content: %s", err)
+			return allowlist
+		}
+
+		for _, entry := range allowlistEntries {
+			rateInfoByQuorum, ok := allowlist[entry.Account]
+			if !ok {
+				allowlist[entry.Account] = map[core.QuorumID]PerUserRateInfo{
+					core.QuorumID(entry.QuorumID): {
+						Name:       entry.Name,
+						Throughput: common.RateParam(entry.ByteRate),
+						BlobRate:   common.RateParam(entry.BlobRate * blobRateMultiplier),
+					},
+				}
+			} else {
+				rateInfoByQuorum[core.QuorumID(entry.QuorumID)] = PerUserRateInfo{
+					Name:       entry.Name,
+					Throughput: common.RateParam(entry.ByteRate),
+					BlobRate:   common.RateParam(entry.BlobRate * blobRateMultiplier),
+				}
+			}
+		}
+	}
+
+	return allowlist
+}
+
+func ReadCLIConfig(c *cli.Context) (RateConfig, error) {
+
+	numQuorums := len(c.IntSlice(RegisteredQuorumFlagName))
+	if len(c.StringSlice(TotalUnauthBlobRateFlagName)) != numQuorums {
+		return RateConfig{}, errors.New("number of total unauth blob rates does not match number of quorums")
+	}
+	if len(c.StringSlice(PerUserUnauthBlobRateFlagName)) != numQuorums {
+		return RateConfig{}, errors.New("number of per user unauth blob intervals does not match number of quorums")
+	}
+	if len(c.IntSlice(TotalUnauthThroughputFlagName)) != numQuorums {
+		return RateConfig{}, errors.New("number of total unauth throughput does not match number of quorums")
+	}
+	if len(c.IntSlice(PerUserUnauthThroughputFlagName)) != numQuorums {
+		return RateConfig{}, errors.New("number of per user unauth throughput does not match number of quorums")
+	}
+
+	quorumRateInfos := make(map[core.QuorumID]QuorumRateInfo)
+	for ind, quorumID := range c.IntSlice(RegisteredQuorumFlagName) {
+
+		totalBlobRate, err := strconv.ParseFloat(c.StringSlice(TotalUnauthBlobRateFlagName)[ind], 64)
+		if err != nil {
+			return RateConfig{}, err
+		}
+		accountBlobRate, err := strconv.ParseFloat(c.StringSlice(PerUserUnauthBlobRateFlagName)[ind], 64)
+		if err != nil {
+			return RateConfig{}, err
+		}
+
+		quorumRateInfos[core.QuorumID(quorumID)] = QuorumRateInfo{
+			TotalUnauthThroughput:   common.RateParam(c.IntSlice(TotalUnauthThroughputFlagName)[ind]),
+			PerUserUnauthThroughput: common.RateParam(c.IntSlice(PerUserUnauthThroughputFlagName)[ind]),
+			TotalUnauthBlobRate:     common.RateParam(totalBlobRate * blobRateMultiplier),
+			PerUserUnauthBlobRate:   common.RateParam(accountBlobRate * blobRateMultiplier),
+		}
+	}
+
+	allowlist := parseAllowlistEntry(c)
```

Comment:
This only reads the entries of the file once right? Don't you want to periodically check the file and update the parameters accordingly?

Reply:
I'll add that in a separate PR